core/tools/generate_names.cc

```diff
@@ -307,8 +307,8 @@ NameDef names[] = {
     {"keepForTypechecking", "keep_for_typechecking"},
     {"unresolvedAncestors", "<unresolved-ancestors>"},
 
-    {"is_a_p", "is_a?"},
-    {"kind_of", "kind_of?"},
+    {"isA_p", "is_a?"},
+    {"kindOf_p", "kind_of?"},
```

Comment:
I think these are the only two left that have non-final underscores; if we changed these, we might be able to add an `ENFORCE` to `generate_names` to make sure only ever add names with a final underscore. I don't know that it'd be the most useful thing in the world, but it also seems like it might be low-hanging fruit?

Reply:
There are actually more:

```
key_p
defined_p
blockGiven_p
nil_p
blank_p
present_p
isA_p
kindOf_p
```

They all line up with ruby methods whose name ends with `?`. I'm led to believe that this is a lisp-ism, which had methods that ended with `-p` for "predicate".

I kind of like that convention.

Reply:
I started to write an ENFORCE for

- no underscores, or
- only terminal `_p`, or
- only terminal `_`

but i realized i didn't like C++'s string manipulation functions so i think i'm going to pass on this. Good suggestion, maybe some other time.